src/storage/mod.rs

```diff
@@ -2015,14 +2069,20 @@ impl<E: Engine, L: LockManager> Storage<E, L> {
                     .get(priority_tag)
                     .inc();
 
-                // TODO: check_api_version
                 if api_version != ApiVersion::V2 {
                     return Err(box_err!("raw_checksum is only available in API V2"));
                 }
                 if algorithm != ChecksumAlgorithm::Crc64Xor {
                     return Err(box_err!("unknown checksum algorithm {:?}", algorithm));
                 }
 
+                let mut keys = Vec::with_capacity(ranges.len() << 1);
+                for range in &ranges {
+                    keys.push(range.get_start_key());
+                    keys.push(range.get_end_key());
+                }
+                Self::check_api_version(api_version, ctx.api_version, CMD, keys)?;
```

Comment:
This may work
```suggestion
                Self::check_api_version(api_version, ctx.api_version, CMD, &[range.get_start_key(), range.get_end_key()])?;
```

Reply:
En...how ? `range` is iterated from `ranges`.